src/test/java/com/google/sps/servlets/LinkServletTest.java

```diff
@@ -0,0 +1,149 @@
+public final class LinkServletTest {
+  @Mock private HttpServletRequest request;
+  @Mock private HttpServletResponse response;
+  private LinkShortenServlet linkShortenServlet;
+  private LinkServlet linkServlet;
+  private String reportsJson;
+  private StringWriter stringWriter;
+  private PrintWriter writer;
+
+  // Maximum eventual consistency.
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig()
+          .setDefaultHighRepJobPolicyUnappliedJobPercentage(100));
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Before
+  public void setUp() throws IOException {
+    MockitoAnnotations.initMocks(this);
+    linkShortenServlet = new LinkShortenServlet();
+    linkServlet = new LinkServlet();
+    helper.setUp();
+    reset();
+    linkServlet.doGet(request, response);
+  }
+
+  /**
+   * Creates new String/PrintWriter and sets status, method, and protocol
+   */
+  private void reset() throws IOException {
+    stringWriter = new StringWriter();
+    writer = new PrintWriter(stringWriter);
+    when(response.getStatus()).thenReturn(HttpURLConnection.HTTP_OK);
+    when(request.getMethod()).thenReturn("get");
+    when(request.getProtocol()).thenReturn("http");
+    when(response.getWriter()).thenReturn(writer);
+  }
+
+  /**
+   * Servlet returns correct contentType, CharacterEncoding, Writer, Headers, and Status
+   */
+  @Test
+  public void servletBehavesCorrectly() throws IOException {
+    verify(response).setContentType("text/html");
+    verify(response).setCharacterEncoding("UTF-8");
+    verify(response).getWriter();
+    Assert.assertEquals(Collections.<String>emptyList(), response.getHeaderNames());
+    Assert.assertEquals(200, response.getStatus());
+  }
+
+  /**
+   * Id parameter not provided
+   */
+  @Test
+  public void noId() throws IOException {
+    reset();
+    linkServlet.doGet(request, response);
+    Assert.assertEquals("Invalid Id\n", stringWriter.toString());
+  }
+
+  /**
+   * Id does not exist
+   */
+  @Test
+  public void invalidId() throws IOException {
+    reset();
+    when(request.getParameter("id")).thenReturn("1");
+    linkServlet.doGet(request, response);
+    Assert.assertEquals("Invalid Id\n", stringWriter.toString());
+  }
+
+  /**
+   * Url stored then retrieved
+   */
+  @Test
+  public void singleRequest() throws IOException {
+    reset();
+    String url = "http://example.com";
+    when(request.getParameter("url")).thenReturn(url);
+    linkShortenServlet.doGet(request, response);
+    String id = stringWriter.toString().replace("\n", "");
+    reset();
+    when(request.getParameter("id")).thenReturn(id);
+    linkServlet.doGet(request, response);
+    Assert.assertEquals(url, stringWriter.toString().replace("\n", ""));
+  }
+
+  /**
+   * Different urls stored then retrieved
+   */
+  @Test
+  public void multipleRequests() throws IOException {
+    final int NUM_REQUESTS = 5;
```

Comment:
Local variables should use lower camel case, see: https://google.github.io/styleguide/javaguide.html#s5.2.7-local-variable-names

If this local variable is only used once within the scope of this function, feel free to remove it.